src/network/network_coordinator.cpp

```diff
@@ -149,6 +202,58 @@ bool ClientNetworkCoordinatorSocketHandler::Receive_GC_LISTING(Packet *p)
 	return true;
 }
 
+bool ClientNetworkCoordinatorSocketHandler::Receive_GC_CONNECTING(Packet *p)
+{
+	std::string token = p->Recv_string(NETWORK_TOKEN_LENGTH);
+	std::string invite_code = p->Recv_string(NETWORK_INVITE_CODE_LENGTH);
+
+	/* Find the connecter based on the invite code. */
+	auto connecter_it = this->connecter_pre.find(invite_code);
+	if (connecter_it == this->connecter_pre.end()) {
+		this->CloseConnection();
+		return false;
+	}
+
+	/* Now store it based on the token. */
+	this->connecter[token] = connecter_it->second;
```

Comment:
Can't the tokens and invite_codes be in the same map? After all, the invite codes always start with `+` so with little effort one could prevent tokens to start with `+` and they could use the same space. After all, in both cases its just a mapping of a string to the connecter.

Reply:
Is there a benefit of having them in the same map?

The reason I used two, is because it makes debugging slightly easier; if something is stuck in `connecter_pre` it means something very different then when it is stuck in `connecter`. It is more about the clarity of what the bookkeeping is doing.

So, can it? Sure. But do we want to?

Reply:
The main benefit would be fewer maps to maintain, though that benefit might be outweighed by the debugging benefit of having several maps. So it's somewhat of an open question where I'm agnostic to the answer.

Reply:
In that case, given the key for both maps really indicate something completely different, I rather keep two maps :) Pretty sure that will help future-us :D